### DISCUSSION TITLE
using the WatchDog Timer (WDT)

### DISCUSSION THREAD

**Discussion by Flipje1955** [micropython/micropython#9322] [Using MicroPython] [answered]
As a possible way to protect valuable hardware against a continued power-on situation, I am looking at the WDT. The use is simple and effective. 
In playing with it, it appears that wdt.feed() method resets the WDT to the value supplied in the constructor. As the documentation is not explicit, is that how it is really designed to work ?

Out of interest and on the same topic, there is no mechanism to read the time remaining in the current WDT setting ?
Thanks !

**Comment by robert-hh** [comment, marked as answer]
The WDT() mechanism varies somewhat between the ports. While to enable it with a certain alarm time, some allow to change the alarm time while operating. But that is hardware dependent.
Which device do you use?

**Comment by Flipje1955** [comment]
Thx for very quick reply. Am using Pico's, so rp2040's and played with it using MicroPython v1.19.1.

Changing the WDT time is an option, but being assured the "feed" essentially resets it to the initial time-out is also fine.

**Reply by robert-hh** [reply]
Yes, feed resets the WDT to the initial time period.
